### PR TITLE
Update OrbitalFrequencies.m

### PR DESCRIPTION
**Pull request by jrivest3**
Update made to definition of KerrGeoMinoFrequencies[a_, p_, e_, x_] that allows handling of the case when x = 0. The math is also simplified overall. I have not analyzed the implications for cases of a = 1 or e = 1.

### CODE REVIEW

File: Kernel/OrbitalFrequencies.m [hunk 1]
```diff
@@ -127,44 +127,133 @@
...
+   Values[KerrGeoConstantsOfMotion[a, p, e, x]];
+  {r1, r2, r3, r4} = KerrGeoRadialRoots[a, p, e, x, En, Q];
+  (*{zminus,zplus}={Sqrt[1-x^2],Sqrt[Q/(1-x^2)]}(* = 
+  KerrGeoPolarRoots[a,p,e,x,En,L]*),
+  {zm,zp}={zminus^2,zplus^2},(*For some reason, 
+  Fujita and Hikida define zm=zminus^2 \[Equal] z1 and zp=
+  Q/(L^2 \[Epsilon]0 zm), with \[Epsilon]0=a^2(1\[Minus]En^2)/L^2, 
+  so that zp \[Equal] zplus^2/(a^2(1-En^2)) \[Equal] z2/(a^2(1-
+  En^2)). *)
+  *)
```
Comment (duetosymmetry) [inline Kernel/OrbitalFrequencies.m]: Please remove all commented-out code.

File: Kernel/OrbitalFrequencies.m [hunk 2]
```diff
@@ -127,44 +127,133 @@
...
+  a L)rm-2M a^2 En)/(r3-rm) (EllipticK[kr^2]-(r2-r3)/(r2-
+  rm) EllipticPi[hm,kr^2])));*)
+  <|"\!\(\*SubscriptBox[\(\[CapitalUpsilon]\), \(r\)]\)" -> \
+\[CapitalUpsilon]r, 
+   "\!\(\*SubscriptBox[\(\[CapitalUpsilon]\), \(\[Theta]\)]\)" -> 
+    Abs[\[CapitalUpsilon]\[Theta]], 
+   "\!\(\*SubscriptBox[\(\[CapitalUpsilon]\), \(\[Phi]\)]\)" -> \
+\[CapitalUpsilon]\[Phi], 
+   "\!\(\*SubscriptBox[\(\[CapitalUpsilon]\), \(t\)]\)" -> \
+\[CapitalGamma]|>
```
Comment (duetosymmetry) [inline Kernel/OrbitalFrequencies.m]: Avoid making whitespace changes unless you're doing whitespace cleanup (and if so then try to do that in a separate commit so it's easy to ignore)

File: Kernel/OrbitalFrequencies.m [hunk 3]
```diff
@@ -127,44 +127,133 @@
...
+  (*{zminus,zplus}={Sqrt[1-x^2],Sqrt[Q/(1-x^2)]}(* = 
+  KerrGeoPolarRoots[a,p,e,x,En,L]*),
+  {zm,zp}={zminus^2,zplus^2},(*For some reason, 
+  Fujita and Hikida define zm=zminus^2 \[Equal] z1 and zp=
+  Q/(L^2 \[Epsilon]0 zm), with \[Epsilon]0=a^2(1\[Minus]En^2)/L^2, 
+  so that zp \[Equal] zplus^2/(a^2(1-En^2)) \[Equal] z2/(a^2(1-
+  En^2)). *)
+  *)
+  z1 = 1 - x^2;
+  If [z1 == 0., z2 = a^2 (1 - En^2) + L^2;, z2 = Q/z1;];(*(a^2 (1-
```
Comment (duetosymmetry) [inline Kernel/OrbitalFrequencies.m]: I think we'd prefer an assignment of the form `z2 = If[test, case1, case2];`.

Also, will this test work for exact 0 instead of floating point 0?

Comment (jrivest3) [reply Kernel/OrbitalFrequencies.m]:
```suggestion
  z2 = If [z1 == 0, a^2 (1 - En^2) + L^2, Q/z1];
```
Seems to work fine with either '0' or '0.'. 
I found that the syntax (0|0.) didn't work there, but it doesn't seem to be necessary.

File: Kernel/OrbitalFrequencies.m (hunk 2 above)
Comment (jrivest3) [reply Kernel/OrbitalFrequencies.m]: I wasn't paying attention to the whitespace. I think that just happened from pasting between the raw file and the Mma notebook.

File: Kernel/OrbitalFrequencies.m (hunk 1 above)
Comment (jrivest3) [reply Kernel/OrbitalFrequencies.m]: I cleaned it up, but I left that bit to explain what I was doing. If you think it's not necessary, it can be removed.

**Comment by nielsw2** [comment]
The polar orbital frequencies were fixed in pull request #48 which has now been merged into the master branch. I think some of those fixes came from looking at this pull request.

Can you check if the master branch now behaves as you expect for polar orbits. If so I will close this pull request.

**Comment by jrivest3** [comment]
Almost. For KerrGeoMinoFrequencyt\[Theta], line 253 of OrbitalFrequancies.m, it looks like an a^2 got dropped from the EllipticK function. I would define k\[Theta]=a^2 (1-En^2)(zm/zp)^2.
Another thing: It's cool to have a solution for UpsilonPhi(x=0) worked out. However, it looks to me like this solution gives the limit as x -> +0 and that the limit as x -> -0 would be different. If that is an intentional choice, it might be worth noting the discontinuity in the documentation/package tutorial.

**Comment by nielsw2** [comment]
Thanks for checking. I'll look into the issue on line 253.

@MvdMeent I think the limit for UpsilonPhi(x=0)  is exactly what we were discussing the other day. Given the limit is discontinuous we probably should return Undefined, or both results in a way that is clear to the user.

**Comment by MvdMeent** [comment]
> Thanks for checking. I'll look into the issue on line 253.
> 
> @MvdMeent I think the limit for UpsilonPhi(x=0) is exactly what we were discussing the other day. Given the limit is discontinuous we probably should return Undefined, or both results in a way that is clear to the user.

While the frequency is discontinuous, the resulting orbit is well-defined in the polar limit. The most important thing is that the frequency generated is what the rest of KerrGeoOrbit expects. Currently, KerrGeoOrbit produces an indeterminate phi coordinate.

**Comment by MvdMeent** [comment]
> Almost. For KerrGeoMinoFrequencyt[Theta], line 253 of OrbitalFrequancies.m, it looks like an a^2 got dropped from the EllipticK function. I would define k[Theta]=a^2 (1-En^2)(zm/zp)^2.

I confirm there is an a^2 missing there. After correcting it, the outputs  of  KerrGeoMinoFrequencyt  agree with the outputs of my code everywhere, including edge cases. (Somehow this function was missed in my previous run verifying the results.

**Comment by jrivest3** [comment]
Whatever the function returns, it might be helpful to inform users of the discontinuity, maybe on the Wolfram Doc/Reference page under 'Details and Options' or in the Examples or something. 
Overall, it looks good to me, and I agree with having this pull request closed.

**Comment by nielsw2** [comment]
I have pushed the fix in KerrGeoMinoFrequencyt[Theta] to the master branch. When we update the documentation we can, as suggested, add a note about the discontinuity in the frequencies. I will close this pull request now but open an issue to sort the phi behaviour of the orbit for polar orbits.
